### PR TITLE
Applications (SP) listing feature for the Admin Portal

### LINKED ISSUE #322: Application Management - Create Basic List View
We should be able to list all the applications in the dev portal with brief intuitive details.

### PR DESCRIPTION
**Pull request by brionmario**
Resolves #322

## Purpose
The applications(service providers) should be listed on the admin portal for the admin user's to manage and configure.

## Goals
This PR brings application listing along with pagination support. Components for filtering and sorting has been placed but aren't yet integrated.

## Approach
A generic component called `ResourceList` and a layout called `ListLayout` has been created so that it could be reused to maintain consistency in listing throughout the admin portal.

Screen Shot 2020-02-03 at 5 16 30 PM

### CODE REVIEW

File: modules/theme/src/themes/default/modules/dropdown.overrides [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NipuniBhagya) [inline modules/theme/src/themes/default/modules/dropdown.overrides]: Shall we change the year?

File: modules/theme/src/themes/default/modules/dropdown.variables [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (NipuniBhagya) [inline modules/theme/src/themes/default/modules/dropdown.variables]: We should change the year here as well.

File: apps/admin-portal/src/components/applications/application-list.tsx [hunk 3]
```diff
@@ -16,110 +16,83 @@
...
-                (
-                    <Grid>
-                        <Grid.Row divided>
-                            <Grid.Column width={ 10 }>
-                                <p>No application found</p>
-                            </Grid.Column>
-                        </Grid.Row>
-                    </Grid>
-                )
+        <AdminList className="applications-list">
```
Comment (NipuniBhagya) [inline apps/admin-portal/src/components/applications/application-list.tsx]: Can we rethink the component name 'AdminList' here?

File: modules/theme/src/themes/default/modules/dropdown.overrides (hunk 1 above)
Comment (brionmario) [reply modules/theme/src/themes/default/modules/dropdown.overrides]: Fixed with commit `2811838`

File: modules/theme/src/themes/default/modules/dropdown.variables (hunk 2 above)
Comment (brionmario) [reply modules/theme/src/themes/default/modules/dropdown.variables]: Fixed with commit `2811838`

File: apps/admin-portal/src/components/applications/application-list.tsx (hunk 3 above)
Comment (brionmario) [reply apps/admin-portal/src/components/applications/application-list.tsx]: Any suggestions?

File: apps/admin-portal/src/locales/en/views.ts [hunk 4]
```diff
@@ -20,6 +20,79 @@ import { Views } from "../../models";
...
+                    title: "No Applications"
+                }
+            },
+            search: {
+                forms: {
+                    searchForm: {
+                        inputs: {
+                            filerAttribute: {
+                                label: "Filter attribute",
+                                placeholder: "ex: name, description etc.",
```
Comment (jeradrutnam) [inline apps/admin-portal/src/locales/en/views.ts]:
```suggestion
                                placeholder: "E.g. name, description etc ...",
```

Do it in other places as well.

Comment (brionmario) [reply apps/admin-portal/src/locales/en/views.ts]: Fixed in `d9385d4`

**Pull request merged**